src/citrine/informatics/predictor_evaluator.py

```diff
@@ -23,28 +24,36 @@ def get_type(cls, data) -> Type[Serializable]:
             "HoldoutSetEvaluator": HoldoutSetEvaluator
         }[data["type"]]
 
-    def _attrs(self) -> List[str]:
-        raise NotImplementedError  # pragma: no cover
-
     def __eq__(self, other):
-        try:
-            return all([
-                self.__getattribute__(key) == other.__getattribute__(key) for key in self._attrs()
-            ])
-        except AttributeError:
+        if isinstance(other, Serializable):
+            self_dict = self.dump()
+            other_dict = other.dump()
+
+            self_dict['responses'] = set(self_dict.get('responses', []))
+            self_dict['metrics'] = frozenset(
+                frozenset((k, v) for k, v in dct.items()) for dct in self_dict.get('metrics', [])
+            )
```

Comment:
A little gross and could use a regular set instead of a frozenset on the outside, but structurally sound.  Can't just use the metrics hashing function because CoverageProbability compares the levels even though its hashing function ignores that field.